### PR TITLE
Add formatter setting for indenting where clauses

### PR DESCRIPTION
**Pull request by msmorgan**
This allows the intellij-rust formatter to better approximate the rustfmt formatter,
which by default aligns where clauses with the parent declaration.

### CODE REVIEW

**Comment by Undin** [comment]
@msmorgan I tried both stable and nightly rustfmt and it doesn't align `where` clause in this way
Current difference between rustfmt and our formatter that rustfmt moves type bounds on new lines and aligns them

**Comment by Undin** [comment]
@msmorgan Oh, sorry, my mistake
This PR is about that **our** formatter indents `where` clauses but rustfmt doesn't

File: src/test/kotlin/org/rust/ide/formatter/RsFormatterTest.kt [hunk 1]
```diff
@@ -90,6 +90,16 @@ class RsFormatterTest : RsFormatterTestBase() {
         """)
     }
 
+    fun `test indent where clause on`() {
+        custom().INDENT_WHERE_CLAUSE = true
+        doTest()
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/formatter/RsFormatterTest.kt]: Let's inline test code from files
It significantly simplifies reading
See `test align where bounds off` test as example

File: src/main/kotlin/org/rust/ide/formatter/settings/RsCodeStyleSettings.kt [hunk 2]
```diff
@@ -15,6 +15,8 @@ class RsCodeStyleSettings(container: CodeStyleSettings) :
     @JvmField var ALIGN_WHERE_CLAUSE = false
     @JvmField var ALIGN_TYPE_PARAMS = false
     @JvmField var ALIGN_WHERE_BOUNDS = true
+    @JvmField
+    var INDENT_WHERE_CLAUSE = true
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/formatter/settings/RsCodeStyleSettings.kt]: Let's place annotation and variable to one line to have the same code style

File: src/test/resources/org/rust/ide/formatter/fixtures/indent_where_clause_off.rs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+impl moo {
+    pub fn do_stuff<T>(&self, item: &T)
+where T: ToString {}
+}
```
Comment (Undin) [inline src/test/resources/org/rust/ide/formatter/fixtures/indent_where_clause_off.rs]: Looks like we can simplify test examples and drop `impl` blocks

**Comment by Undin** [comment]
@msmorgan Do you need any help?

**Comment by msmorgan** [comment]
No, and thanks for the bump. I'd meant to get back to this but forgot.

File: src/test/resources/org/rust/ide/formatter/fixtures/indent_where_clause_off.rs (hunk 3 above)
Comment (msmorgan) [reply src/test/resources/org/rust/ide/formatter/fixtures/indent_where_clause_off.rs]: I've done what you asked. The test above mine had the impl block so I did the same. It adds some visual clarity to the expected indentation, but it isn't strictly necessary.

File: src/main/kotlin/org/rust/ide/formatter/settings/RsCodeStyleSettings.kt (hunk 2 above)
Comment (msmorgan) [reply src/main/kotlin/org/rust/ide/formatter/settings/RsCodeStyleSettings.kt]: I think I was punked here by IntelliJ's "reformat code" setting in the commit dialog.

**Comment by Undin** [comment]
Great 👍 
Thanks!
bors r+

**Pull request merged**